### PR TITLE
NSA-3195

### PR DESCRIPTION
**Pull request by jovin-joy**

### CODE REVIEW

File: src/infrastructure/Accounts/index.js [hunk 1]
```diff
@@ -31,17 +32,19 @@ class Account {
...
       const userDirectoriesUrl = `${config.accounts.url}users/${id}`;
       let correlationId = uuid();
-
       if (ctx && ctx.req && ctx.req.id) {
-        correlationId = ctx.req.id;
+          correlationId = ctx.req.id;
       }
-
-      logger.info(`calling directories api with: ${userDirectoriesUrl}`);
+      const cacheResponse = cache.retrieve(userDirectoriesUrl);
```
Comment (DrogoNevets) [inline src/infrastructure/Accounts/index.js:42]: the key should really be just the userId no a URL

File: src/infrastructure/Accounts/index.js [hunk 2]
```diff
@@ -53,20 +56,35 @@ class Account {
...
         if (!response.body) {
           logger.warn(`user not found Id:${id}`);
           return null;
         }
-        const user = response.body;
-        logger.info(`user returned : ${user}`);
-        returnValue = user === undefined ? null : user;
+        user = response.body;
+        const isStored = cache.save(userDirectoriesUrl, user);
+        if(isStored !== true){
```
Comment (DrogoNevets) [inline src/infrastructure/Accounts/index.js:66]: do we know why it wasnt stored?

File: src/infrastructure/helpers/AppCache.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+const NodeCache = require('node-cache');
```
Comment (DrogoNevets) [inline src/infrastructure/helpers/AppCache.js:1]: lets make this file return a service class instance and go down the OOP route

File: test/infrastructure/AccountsTests/index.test.js [hunk 4]
```diff
@@ -48,6 +50,7 @@ describe('When constructing the accounts', () => {
     req: { id: correlationId }
   };
   beforeEach(() => {
+      cache.flushAll();
```
Comment (DrogoNevets) [inline test/infrastructure/AccountsTests/index.test.js:53]: we need to test the cache too please

Comment (jovin-joy) [reply test/infrastructure/AccountsTests/index.test.js:53]: it will test with each test, this line is to flush before running next test task

Comment (jovin-joy) [reply test/infrastructure/AccountsTests/index.test.js:53]: the test classes will cover the cache test as well.

File: src/infrastructure/Accounts/index.js (hunk 1 above)
Comment (jovin-joy) [reply src/infrastructure/Accounts/index.js:42]: Done

File: test/infrastructure/AccountsTests/index.test.js (hunk 4 above)
Comment (DrogoNevets) [reply test/infrastructure/AccountsTests/index.test.js:53]: but we dont have a unit test for the actual cahce

File: test/infrastructure/helpersTests/AppCache.test.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            "family_name": "Joy",
+            "email": "test@dfe.com",
+            "status": 1,
+            "phone_number": null,
+            "isMigrated": false
+        });
+    });
+
+    it("Should flush all contents from cache", async () => {
+        await cache.flushAll(); // It Just flush all after running above tests.
```
Comment (DrogoNevets) [inline test/infrastructure/helpersTests/AppCache.test.js:49]: there ar eno tests here

Comment (DrogoNevets) [inline test/infrastructure/helpersTests/AppCache.test.js:49]: no test to check TTL

**Pull request merged**